peachjam/resources.py

```diff
@@ -146,31 +163,51 @@ def before_import_row(self, row, **kwargs):
                 defaults={"name": row["court_obj"]["name"]},
             )
 
+        source_files = row["source_url"].split("|")
+        docx = re.compile(r".docx?$")
+        pdf = re.compile(r".pdf$")
+        for file in source_files:
+            match_docx = docx.search(file)
+            match_pdf = pdf.search(file)
+            if match_docx:
+                row["source_url"] = file
+            elif match_pdf:
+                row["source_url"] = file
+
     def after_import_row(self, row, instance, row_number=None, **kwargs):
         super().after_import_row(row, instance, row_number, **kwargs)
 
         judgment = Judgment.objects.get(pk=instance.object_id)
 
-        if row["judges"]:
-            for judge in list(map(str.strip, row["judges"].split("|"))):
-                judge, _ = Judge.objects.get_or_create(name=judge)
-                judgment.judges.add(judge)
-
-        if "string_override" in row:
+        if row["case_string_override"]:
             CaseNumber.objects.create(
-                string_override=row["string_override"], document=judgment
+                string_override=row["case_string_override"], document=judgment
             )
-        elif "case_numbers" in row:
-            for case_number in list(map(str.strip, row["case_numbers"].split("|"))):
-                if row["matter_type"]:
-                    matter_type, _ = MatterType.objects.get_or_create(
-                        name=row["matter_type"]
-                    )
-                CaseNumber.objects.create(
-                    document=judgment,
-                    number=case_number[0],
-                    year=case_number[1],
-                    matter_type=MatterType.objects.get(name=row["matter_type"]),
+        elif row["case_numbers"]:
+            for c in list(map(str.strip, row["case_numbers"].split("|"))):

```

Comment:
Can you include a comment showing the type of input you're expecting?

Reply:
added.